archunit/src/main/java/com/tngtech/archunit/ArchConfiguration.java

```diff
@@ -311,16 +310,23 @@ void remove(String propertyName) {
         }
 
         Properties getMergedProperties() {
-            Properties result = createProperties(this.baseProperties);
-            Properties overwritten = getSubProperties("archunit", System.getProperties());
-
-            result.putAll(overwritten);
-            if (!this.finalProperties.equals(result)) {
-                LOG.info("Merging properties: The following properties have been overwritten by system properties: {}", overwritten);
-                this.finalProperties = result;
+            Properties result = createProperties(baseProperties);
+            Properties currentlyOverwritten = getSubProperties("archunit", System.getProperties());
+            result.putAll(currentlyOverwritten);
+
+            if (!overwrittenProperties.equals(currentlyOverwritten)) {
```

Comment:
I'm unsure about the goal of the `overwrittenProperties` field. Is it used to compare against no overrides vs what's currently available on system overrides. Would it make sense to update the field?

Reply:
`overwrittenProperties` is there to only log on change :thinking: I.e. if you request properties a second time, but you still have the same `overwrittenProperties` (i.e. `overwrittenProperties` == `currentlyOverwritten`) then we don't want to log, right? After checking you could theoretically always update `overwrittenProperties`, but if they are equal anyway, it is unnecessary, isn't it? If not equal, then `overwrittenProperties` is updated in the next line...
I hope I understood your question? Do you think it would be easier to understand, if we would always update `overwrittenProperties`, even if they are equal to `currentlyOverwritten` anyways?